empress_gui.py

```diff
@@ -730,13 +748,29 @@ def compute_recon_solutions(self):
             for i in range(len(clusters)):
                 App.medians.append(clusters[i].median())
 
-        if self.view_solution_space_window is not None and self.view_solution_space_window.winfo_exists():
-            self.view_solution_space_window.destroy()
-        if self.view_reconciliations_window is not None and self.view_reconciliations_window.winfo_exists():
-            self.view_reconciliations_window.destroy()
+        if self.view_solution_space_window_after_setting_clusters is not None and self.view_solution_space_window_after_setting_clusters.winfo_exists():
+            self.view_solution_space_window_after_setting_clusters.destroy()
+
+        if self.num_cluster is not None:
+            for i in range(self.num_cluster):
+                if self.solution_window is not None and self.solution_window.winfo_exists():
```

Comment:
It seems like you are iterating from `i = 0` to `i = self.num_cluster - 1` but you never used the value `i`.

Reply:
Yes, because I wanted to delete the self.num_cluster number of tkinter windows created by "One per cluster".

Reply:
Seems like you are destroying the same `self.solution_window` multiple times.

Reply:
I have removed it.